src/renderer/components/kube-object-menu/kube-object-menu-item-injection-token.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectionToken } from "@ogre-tools/injectable";
+import type { IComputedValue } from "mobx";
+import type { KubeObjectMenuProps } from "./kube-object-menu";
+
+export interface KubeObjectMenuItem {
+  kind: string;
+  apiVersions: string[];
+  enabled: IComputedValue<boolean>;
+
+  // TODO: Figure out why "KubeObject" does not work here.
+  Component: React.ElementType<KubeObjectMenuProps<any>>;
```

Comment:
The reason that using `KubeObject` here doesn't work is that the components being registered say that they require `class T extends KubeObject` which is not the same and typescript rightly complains that any arbitrary instance unrelated to `T` could be passed in.

Personally, I would use `KubeObject` here (and factor out this type as:

```ts
export type KubeObjectMenuItemComponent = React.ElementType<KubeObjectMenuProps<KubeObject>>;
```

and then at the uses of it I would add an `as KubeObjectMenuItemComponent` cast.

Reply:
Changed.